app/src/main/java/com/example/marvelapp/api/service/CharacterServiceImpl.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.marvelapp.api.service
+
+import com.example.marvelapp.api.MarvelAppApi
+import com.example.marvelapp.api.mapper.CharacterMapper
+import com.example.marvelapp.api.util.Result
+import com.example.marvelapp.entity.Character
+
+class CharacterServiceImpl : CharacterService {
+
+    private val api = ServiceGenerator()
```

Comment:
this should be passed by parameter, never created here as it is violating the "D" from SOLID

Reply:
done